File: src/main/java/aurocosh/divinefavor/common/spirit/ModSpirits.java
package aurocosh.divinefavor.common.spirit;

import aurocosh.divinefavor.common.config.common.ConfigSpirits;
import aurocosh.divinefavor.common.spirit.base.ModSpirit;
import aurocosh.divinefavor.common.spirit.punishment.*;

public final class ModSpirits {
    public static ModSpirit arbow;
    public static ModSpirit blizrabi;
    public static ModSpirit endererer;
    public static ModSpirit loon;
    public static ModSpirit neblaze;
    public static ModSpirit redwind;
    public static ModSpirit romol;
    public static ModSpirit squarefury;
    public static ModSpirit timber;

    public static void preInit() {
        arbow = new ModSpirit("arbow", new SquareFuryPunishment(), ConfigSpirits.arbow);
        blizrabi = new ModSpirit("blizrabi", new BlizrabiPunishment(), ConfigSpirits.blizrabi);
        endererer = new ModSpirit("endererer", new EnderererPunishment(), ConfigSpirits.endererer);
        loon = new ModSpirit("loon", new LoonPunishment(), ConfigSpirits.loon);
        neblaze = new ModSpirit("neblaze", new NeblazePunishment(), ConfigSpirits.neblaze);
        redwind = new ModSpirit("redwind", new RedwindPunishment(), ConfigSpirits.redwind);
        romol = new ModSpirit("romol", new RomolPunishment(), ConfigSpirits.romol);
        squarefury = new ModSpirit("squarefury", new SquareFuryPunishment(), ConfigSpirits.squarefury);
        timber = new ModSpirit("timber", new TimberPunishment(), ConfigSpirits.timber);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/spirit/ModSpirits.kt
package aurocosh.divinefavor.common.spirit

import aurocosh.divinefavor.common.config.common.ConfigSpirits
import aurocosh.divinefavor.common.spirit.base.ModSpirit
import aurocosh.divinefavor.common.spirit.punishment.*

object ModSpirits {
    lateinit var arbow: ModSpirit
    lateinit var blizrabi: ModSpirit
    lateinit var endererer: ModSpirit
    lateinit var loon: ModSpirit
    lateinit var materia: ModSpirit
    lateinit var neblaze: ModSpirit
    lateinit var redwind: ModSpirit
    lateinit var romol: ModSpirit
    lateinit var squarefury: ModSpirit
    lateinit var timber: ModSpirit

    fun preInit() {
        arbow = ModSpirit("arbow", ArbowPunishment(), ConfigSpirits.arbow)
        blizrabi = ModSpirit("blizrabi", BlizrabiPunishment(), ConfigSpirits.blizrabi)
        endererer = ModSpirit("endererer", EnderererPunishment(), ConfigSpirits.endererer)
        loon = ModSpirit("loon", LoonPunishment(), ConfigSpirits.loon)
        materia = ModSpirit("materia", LoonPunishment(), ConfigSpirits.materia) // TODO replace punishment
        neblaze = ModSpirit("neblaze", NeblazePunishment(), ConfigSpirits.neblaze)
        redwind = ModSpirit("redwind", RedwindPunishment(), ConfigSpirits.redwind)
        romol = ModSpirit("romol", RomolPunishment(), ConfigSpirits.romol)
        squarefury = ModSpirit("squarefury", SquareFuryPunishment(), ConfigSpirits.squarefury)
        timber = ModSpirit("timber", TimberPunishment(), ConfigSpirits.timber)
    }
}